nexus/src/app/sagas/disk_create.rs

```diff
@@ -581,7 +581,7 @@ fn randomize_volume_construction_request_ids(
 }
 
 #[cfg(test)]
-mod test {
+pub(crate) mod test {
```

Comment:
I'm mostly modifying this so I can re-use some of the test functions which validates "disk creation doesn't leave detritus".

Most of these checks are identical for snapshot creation.